packages/generate/src/cli.ts

```diff
@@ -238,6 +238,9 @@ const run = async () => {
       case "--no-update-ignore-file":
         options.updateIgnoreFile = false;
         break;
+      case "--future":
+        options.future = true;
```

Comment:
I don't think this is sufficient. We need to be able to specify which futures you are opting into. I could be ok with future A and have updated my codebase to account for it. But then future B is added and I would be automatically opted into it as well which is a breaking change I wasn't ready for.

Reply:
Yeah, @diksipav and I talked a bit about this yesterday and here are my thoughts:

1. From the perspective of a general feature of the generator package, this `future` behavior is similar to TypeScript's own `strict` behavior, which if you opt into directly is breaking between versions. It stands for some set of more granular configuration values, so I think this _is_ useful in general.
2. From the perspective of this future behavior, I expect most (all?) folks to want to opt into both, but also would be fine exposing granular flags for each here, too. I don't want to add tests for each permutation since we plan on making this the _only_ behavior in a future breaking change, but that does mean there might be bugs or poorly understood side-effects of having one or the other enabled, but not both.

I don't think we need to block on this, though, and I'm fine with trying to land this as-is now. I'll leave it up to @diksipav to decide if she wants to address this in this PR specifically, or as a follow-up PR.

Reply:
hey lets merge this (once it's ready) and I'll create right away a follow up PR with individual future flags

Reply:
@scotttrinh I'll merge this and start working on the PR with individual flags, if u think we should "swap" tests will do it in the next PR